### PR TITLE
Ubuntu 16.04 support for **just** GZIP ManyBugs image 

### PR DESCRIPTION
**Pull request by pdreiter**
The following are the subset of changes specified by @ChrisTimperley in pull request #18 - in order to fix issue #17. 

1. Ubuntu 16.04 and manybugs:base Dockerfile contents have been prepended to gzip/Dockerfile - required base/preprocess and compile.sh to be added to gzip dir
2. Reverted perlbrew perl version installation to original Dockerfile construct because after local testing, it did work - not sure what I might have broken in my earlier dev.  
3. Specifically reference 3.6.7 bootstrap version

### CODE REVIEW

File: gzip/Dockerfile [hunk 1]
```diff
@@ -1,6 +1,74 @@
-FROM squareslab/manybugs:base
+FROM ubuntu:16.04
 MAINTAINER Chris Timperley "christimperley@gmail.com"
 
+#---------------------------------
```
Comment (ChrisTimperley) [inline gzip/Dockerfile]: It's fine to drop these head/tail comments.

**Comment by pdreiter** [comment]
> This looks good. I think that you could make one small change to reduce the amount of code duplication: if you move `gzip/Dockerfile` to `Dockerfile.gzip` at the root of the repo, then you shouldn't need to create copies of existing files.

If I'm not wrong, this change also means that the gzip.bugzoo.yml needs to be updated:
```
  3 blueprints:
  4   - type: docker
  5     tag: squareslab/manybugs:gzip
  6     file: Dockerfile.gzip
  7     depends-on: squareslab/manybugs:base
```
right?

**Comment by ChrisTimperley** [comment]
> > This looks good. I think that you could make one small change to reduce the amount of code duplication: if you move `gzip/Dockerfile` to `Dockerfile.gzip` at the root of the repo, then you shouldn't need to create copies of existing files.
> 
> If I'm not wrong, this change also means that the gzip.bugzoo.yml needs to be updated:
> 
> ```
>   3 blueprints:
>   4   - type: docker
>   5     tag: squareslab/manybugs:gzip
>   6     file: Dockerfile.gzip
>   7     depends-on: squareslab/manybugs:base
> ```
> 
> right?
> is there any impact to bugzoo when referring to dockerfiles not in the current working directory?

That's right. It's probably best to remove that existing blueprint and to add its equivalent to `manybugs.bugzoo.yml`:
https://github.com/squaresLab/ManyBugs/blob/master/manybugs.bugzoo.yml

**Comment by pdreiter** [comment]
> > > This looks good. I think that you could make one small change to reduce the amount of code duplication: if you move `gzip/Dockerfile` to `Dockerfile.gzip` at the root of the repo, then you shouldn't need to create copies of existing files.
> > 
> > 
> > If I'm not wrong, this change also means that the gzip.bugzoo.yml needs to be updated:
> > ```
> >   3 blueprints:
> >   4   - type: docker
> >   5     tag: squareslab/manybugs:gzip
> >   6     file: Dockerfile.gzip
> >   7     depends-on: squareslab/manybugs:base
> > ```
> > 
> > 
> > right?
> > is there any impact to bugzoo when referring to dockerfiles not in the current working directory?
> 
> That's right. It's probably best to remove that existing blueprint and to add its equivalent to `manybugs.bugzoo.yml`:
> https://github.com/squaresLab/ManyBugs/blob/master/manybugs.bugzoo.yml

Just to be crystal clear, this change requires not only the `-type:` contents, but also the `arguments:` contents of gzip.bugzoo.yml from the blueprints section to be added to manybugs.bugzoo.yml blueprints section. Correct?

**Comment by ChrisTimperley** [comment]
> Just to be crystal clear, this change requires not only the `-type:` contents, but also the `arguments:` contents of gzip.bugzoo.yml from the blueprints section to be added to manybugs.bugzoo.yml blueprints section. Correct?

Not quite. You only need to move the blueprint for the base image to `manybugs.bugzoo.yml`. The blueprints for the individual images should still live in `gzip.bugzoo.yml`.

**Pull request merged**